lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -2517,6 +2517,58 @@ def test_courseware_access(self, waffle_override, course_visibility, user_type,
                     self.assertIn('xblock-student_view-html', response.content)
                     self.assertIn('xblock-student_view-video', response.content)
 
+    @patch('openedx.core.djangoapps.util.user_messages.PageLevelMessages.register_warning_message')
+    def test_courseware_messages_masters_only(self, patch_register_warning_message):
+        with patch('courseware.views.views.CourseTabView.should_show_enroll_button') as patch_should_show_enroll_button:
+            course = CourseFactory()
+
+            user = self.create_user_for_course(course, CourseUserType.UNENROLLED)
+            request = RequestFactory().get('/')
+            request.user = user
+
+            from openedx.core.djangolib.markup import HTML, Text
+            expected_markup_without_button = Text('You must be enrolled in the course to see course content.')
+
+            expected_markup_with_button = u'You must be enrolled in the course to see course content. \
```

Comment:
Is there a way we can test what we want to test here without duplicating this string? It makes things very brittle; we don't really want to test that the copy hasn't changed.

Reply:
I completely agree and was not pleased with it. I had some difficulty doing that the first time around, which is why I went with this. I'll give it a quick shot again.